docs/source/ops/connect/auth0.rst

```diff
@@ -0,0 +1,657 @@
+prior to 1.17.0, you may need to adapt your ``ui/src/config.ts`` and
+``ui/src/components/LoginScreen.tsx`` files. See
+`this commit <https://github.com/digital-asset/daml/commit/79080839c1ca299972038ba515b98e6176668783>`_
+for guidance.
+
+Next, we need to start the Daml driver. For this example we'll use the sandbox,
+but with ``--implicit-party-allocation false`` it should behave like a
+production ledger (minus persistence).
+
+.. code-block:: bash
+
+    cd my-project
+    daml build
+    daml codegen js .daml/dist/my-project-0.1.0.dar -o ui/daml.js
+    daml sandbox --ledgerid %%LEDGER_ID%% \
+                 --auth-jwt-rs256-jwks https://%%AUTH0_DOMAIN%%/.well-known/jwks.json \
+                 --implicit-party-allocation false \
+                 .daml/dist/my-project-0.1.0.dar
+
+As before, you need to replace ``%%LEDGER_ID%%`` with a value of your choosing
+(the same one you used when configuring Auth0), and ``%%AUTH0_DOMAIN%%`` with
+your Auth0 domain, which you can find as the Domain field at the top of the
+Settings tab for any app in the tenant.
+
+Next, you need to start a JSON API instance.
+
+.. code-block:: bash
+
+    cd my-project
+    daml json-api --ledger-port 6865 \
+                  --ledger-host localhost \
+                  --http-port 4000
+
+If you are using a Daml SDK version prior to 1.17.0, you'll need to find a way
+to supply the JSON API with a valid, refreshing token file. We recommend
+upgrading to 1.17.0 or later.
+
+Then, we want to start the Trigger Service and OAuth2 middleware, which we will
+put respectively under ``/trigger`` and ``/auth``. First, the middleware:
+
+.. code-block:: bash
+
+    DAML_CLIENT_ID=%%OAUTH_APP_ID%% \
+    DAML_CLIENT_SECRET=%%OAUTH_APP_SECRET \
+    daml oauth2-middleware \
+      --address localhost \
+      --http-port 5000 \
+      --oauth-auth "https://%%AUTH0_DOMAIN%%/authorize" \
+      --oauth-token "https://%%AUTH0_DOMAIN%%/oauth/token" \
+      --auth-jwt-rs256-jwks "https://%%AUTH0_DOMAIN%%/.well-known/jwks.json" \
+      --callback %%ORIGIN%%/auth/cb
+
+where, as before, you need to replace:
+
+- ``%%OAUTH_APP_ID%%`` with the Client ID value you can find at the top of the
+  settings tab for the OAUTH_APP we just created.
+- ``%%OAUTH_APP_SECRET%%`` with the Client Secret value you can find at the top
+  of the settings tab for the OAUTH_APP we just created.
+- ``%%AUTH0_DOMAIN%%`` with your tenant domain.
+- ``%%ORIGIN%%`` with the full domain-name-or-ip & port, including scheme,
+  under which you expose your server.
+
+Now, the trigger service:
+
+.. code-block:: bash
+
+    daml trigger-service \
+      --address localhost \
+      --http-port 6000 \
+      --ledger-host localhost \
+      --ledger-port 6865 \
+      --auth-internal http://localhost:5000 \
+      --auth-external %%ORIGIN%%/auth \
+      --auth-callback %%ORIGIN%%/trigger/cb \
+      --dar .daml/dist/my-ptoject-0.1.0.dar
```

Comment:
typo `ptoject` should probably be `project`

Reply:
God catch, thanks!